### PR TITLE
Move dynamic library location related logic into wasmer_locator.dart

### PR DESCRIPTION
**Pull request by modulovalue**
This cleans up the dynamic library discovery logic.

In many cases (e.g. where sophisticated build systems are not available) it is much more practical to have a single globally built dynamic library, and to reuse that dynamic library everywhere where it is needed. This doesn't make that possible yet (due to the runtime singleton), but it moves this package in the right direction UX-wise by decoupling the default locations from the actual implementation.

### CODE REVIEW

File: wasm/lib/src/wasmer_locator.dart [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:ffi';
+import 'dart:io';
+
+/// Attempts to locate the Wasmer dynamic library.
+///
+/// Throws [WasmLocatorError] if the dynamic library could not be found.
+DynamicLibrary loadWasmerDynamicLibrary() {
+  String? locate(String libName) {
```
Comment (liamappelbe) [inline wasm/lib/src/wasmer_locator.dart:12]: You can eliminate a lot of the duplicated code below if you have this function return a non-null String, and throw the WasmLocatorError in here.

Comment (modulovalue) [reply wasm/lib/src/wasmer_locator.dart:12]: I agree, but notice how iOS and Android can't fail to be located and how each error message refers to the platform it belongs to.

Handling those that can fail in a separate if statement would remove the clarity that we get here from being explicit. Furthermore, this allows us to easily see that we are being exhaustive with out platform checks and are essentially done here. 

So I think, even if there's some duplication here, for the purposes of clearly communicating the intent of this code to the reader, it should be tolerated.

Comment (liamappelbe) [reply wasm/lib/src/wasmer_locator.dart:12]: You can still have this chain of if statements, so iOS and Android loading would look the same, and you still get the code clarity. I'm just saying throw the exception inside `locate`. The only thing you loose is telling the user what platform they're on, which they presumably already know.

Comment (modulovalue) [reply wasm/lib/src/wasmer_locator.dart]: I've applied your suggested changes.

**Review by liamappelbe — APPROVED** [review]

**Comment by modulovalue** [comment]
macos 2.12 passed now

**Pull request merged**